include/seqan3/io/record.hpp

```diff
@@ -268,6 +272,436 @@ struct record : detail::transfer_template_args_onto_t<field_types, std::tuple>
+    {
+        return get_impl(field_constant<seqan3::field::id>{}, static_cast<base_type const &&>(*this));
+    }
+    //!\copydoc seqan3::record::id
+    decltype(auto) id() &
+    {
+        return get_impl(field_constant<seqan3::field::id>{}, static_cast<base_type &>(*this));
+    }
+    //!\copydoc seqan3::record::id
+    decltype(auto) id() const &
+    {
+        return get_impl(field_constant<seqan3::field::id>{}, static_cast<base_type const &>(*this));
+    }
+
+    //!\brief The "sequence", usually a range of nucleotides or amino acids.
+    decltype(auto) sequence() &&
+    {
+        return get_impl(field_constant<seqan3::field::seq>{}, static_cast<base_type &&>(*this));
+    }
+    //!\copydoc seqan3::record::sequence
+    decltype(auto) sequence() const &&
+    {
+        return get_impl(field_constant<seqan3::field::seq>{}, static_cast<base_type const &&>(*this));
+    }
+    //!\copydoc seqan3::record::sequence
+    decltype(auto) sequence() &
+    {
+        return get_impl(field_constant<seqan3::field::seq>{}, static_cast<base_type &>(*this));
+    }
+    //!\copydoc seqan3::record::sequence
+    decltype(auto) sequence() const &
+    {
+        return get_impl(field_constant<seqan3::field::seq>{}, static_cast<base_type const &>(*this));
+    }
+
+    //!\brief The qualities, usually in phred-score notation.
+    decltype(auto) qualities() &&
+    {
+        return get_impl(field_constant<seqan3::field::qual>{}, static_cast<base_type &&>(*this));
+    }
+    //!\copydoc seqan3::record::qualities
+    decltype(auto) qualities() const &&
+    {
+        return get_impl(field_constant<seqan3::field::qual>{}, static_cast<base_type const &&>(*this));
+    }
+    //!\copydoc seqan3::record::qualities
+    decltype(auto) qualities() &
+    {
+        return get_impl(field_constant<seqan3::field::qual>{}, static_cast<base_type &>(*this));
+    }
+    //!\copydoc seqan3::record::qualities
+    decltype(auto) qualities() const &
+    {
+        return get_impl(field_constant<seqan3::field::qual>{}, static_cast<base_type const &>(*this));
+    }
+
+    //!\brief Sequence (SEQ) relative start position (0-based), unsigned value.
+    decltype(auto) offset() &&
+    {
+        return get_impl(field_constant<seqan3::field::offset>{}, static_cast<base_type &&>(*this));
+    }
+    //!\copydoc seqan3::record::offset
+    decltype(auto) offset() const &&
+    {
+        return get_impl(field_constant<seqan3::field::offset>{}, static_cast<base_type const &&>(*this));
+    }
+    //!\copydoc seqan3::record::offset
+    decltype(auto) offset() &
+    {
+        return get_impl(field_constant<seqan3::field::offset>{}, static_cast<base_type &>(*this));
+    }
+    //!\copydoc seqan3::record::offset
+    decltype(auto) offset() const &
+    {
+        return get_impl(field_constant<seqan3::field::offset>{}, static_cast<base_type const &>(*this));
+    }
+
+    //!\brief Base pair probability matrix of interactions, usually a matrix of float numbers.
```

Comment:
@joergi-w Do you know why it is a `matrix`? From what I can see it is a vector that matches the sequence, i.e. each letter in a sequence has one numerical value.

In the test you called it interactions, would `sequence_interactions` be a good name?

Furthermore, in the test the type of the `base_pair_probability` is `std::vector<std::vector<uint8_t>>` (at least for vienna) and not float.

I'm slightly confused.

Reply:
> @joergi-w Do you know why it is a `matrix`? From what I can see it is a vector that matches the sequence, i.e. each letter in a sequence has one numerical value.

It is a matrix, because each letter in a sequence is paired with every other letter in the same sequence. The float values are the probability of each pairing.
 
> In the test you called it interactions, would `sequence_interactions` be a good name?

An interaction is a 1:1 pairing while the whole matrix describes many:many pairings. A compromise name is interaction probabilities.

> Furthermore, in the test the type of the `base_pair_probability` is `std::vector<uint8_t>` (at least for vienna) and not float.
> 
> I'm slightly confused.

Probably I multiply each value with some factor to work in int-space... I can have a deeper look this afternoon.

Reply:
Thank you for the answer!

> Probably I multiply each value with some factor to work in int-space... I can have a deeper look this afternoon.

That would be awesome :)

> It is a matrix, because each letter in a sequence is paired with every other letter in the same sequence. The float values are the probability of each pairing.

This sound like, if the sequence has size n, then you have a mapping [n] x [n] ↦ float. That would mean n² values, but in the test data we only have n values.

One example is:

```
> example 2
UUGGAGUACACAACCUGUACACUCUUUC
..(((((..(((...)))..)))))... (-3.71)
```

Hmm okay, I have looked at the wrong place, bpp has the following type:

```
std::vector<std::set<std::pair<double, size_t>>> & bpp
```

Do you still know why a hash map wasn't enough as data-structure?

I see, so format_vienna does not only give you the sequence, pseudoknot structure and energy, but also computes that bpp-matrix. In spirit how the alignment is computed from the other values.

Reply:
> Do you still know why a hash map wasn't enough as a data-structure?

Yes, I need access in the order of their probability (the double type value) and this is not a unique key for a hash map. 

> I see, so format_vienna does not only give you the sequence, pseudoknot structure and energy, but also computes that bpp-matrix. In spirit how the alignment is computed from the other values.

Exactly. It becomes clearer and more useful, as soon as we implement further structure formats. But in order to transform LaRA 2 to SeqAn3 I need to wait for pairwise alignment with position-specific score matrices (pssm), so there is low priority for the other formats, currently.

Reply:
> I need access in the order of their probability (the double type value)

Actually with `set<pair>` I wanted to implement a _priority queue_, and this was the best way.